libs/beacon/include/beacon/block_entropy.hpp

```diff
@@ -28,27 +28,35 @@ namespace beacon {
 
 struct BlockEntropy : public BlockEntropyInterface
 {
-  using MuddleAddress     = byte_array::ConstByteArray;
-  using GroupPublicKey    = std::string;
-  using MemberPublicKey   = byte_array::ConstByteArray;
-  using MemberSignature   = byte_array::ConstByteArray;
-  using Confirmations     = std::map<MemberPublicKey, MemberSignature>;
-  using GroupSignature    = dkg::BeaconManager::Signature;
-  using GroupSignatureStr = std::string;
-  using Cabinet           = std::set<MuddleAddress>;
+  using MuddleAddress   = byte_array::ConstByteArray;
+  using GroupPublicKey  = byte_array::ConstByteArray;
+  using MemberPublicKey = byte_array::ConstByteArray;
+  using MemberSignature = byte_array::ConstByteArray;
+  using Confirmations   = std::map<MemberPublicKey, MemberSignature>;
+  using GroupSignature  = byte_array::ConstByteArray;
+  using Cabinet         = std::set<MuddleAddress>;
 
   BlockEntropy();
+  BlockEntropy(BlockEntropy const &rhs);
 
-  Cabinet qualified;  // The members who succeeded DKG and are qualified to produce blocks (when new
-                      // committee)
-  GroupPublicKey group_public_key;  // The group public key (when new committee)
-  uint64_t       block_number = 0;  // The block this is relevant to
-  Digest digest;  // The hash of the above (when new committee) note, this could be implicit. Is not
-                  // serialized.
+  // The members who succeeded DKG and are qualified to produce blocks (when new committee)
+  Cabinet qualified;
 
-  Confirmations confirmations;  // In the case of a new cabinet, personal signatures of the hash
-                                // from qual members
-  GroupSignatureStr group_signature;  // Signature of the previous entropy, used as the entropy
+  // The group public key (when new committee)
+  GroupPublicKey group_public_key;
+
+  // The block this is relevant to
+  uint64_t block_number = 0;
+  // The hash of the above (when new committee) note, this could be implicit. Is not
+  // serialized.
+  Digest digest;
+
+  // In the case of a new cabinet, personal signatures of the hash
+  // from qual members
+  Confirmations confirmations;
+
+  // Signature of the previous entropy, used as the entropy
+  GroupSignature group_signature = GroupSignature{};
```

Comment:
```suggestion
  GroupSignature group_signature{};
```

Reply:
If needed at all?

Reply:
A relic from when I tried to use the MCL stuff direct (a mistake). Good spot will change.